core/src/test/scala/org/apache/spark/scheduler/OutputCommitCoordinatorSuite.scala

```diff
@@ -144,13 +144,6 @@ class OutputCommitCoordinatorSuite extends SparkFunSuite with BeforeAndAfter {
     assert(tempDir.list().size === 1)
   }
 
-  test("If commit fails, if task is retried it should not be locked, and will succeed.") {
```

Comment:
do you know why it is flaky?

Reply:
> do you know why it is flaky?

The event in DAGScheduler is not stable,  TaskEnd of attempt 1 sometime's coming earlier then attempt 0